.github/workflows/ci.yml

```diff
@@ -422,8 +422,9 @@ jobs:
         shell: bash
         run: |
           git config --global core.autocrlf input
-          echo "VCVARSALL=$(vswhere -products \* -2019 -property installationPath)\\VC\\Auxiliary\\Build\\vcvarsall.bat" >> $GITHUB_ENV
-          python_dir=$(cat "/proc/${{ matrix.cygreg }}/HKEY_LOCAL_MACHINE/SOFTWARE/Python/PythonCore/${PYTHON_VER_DOT}/InstallPath/@")
+          echo "VCVARSALL=$(vswhere -products \* -latest -property installationPath)\\VC\\Auxiliary\\Build\\vcvarsall.bat" >> $GITHUB_ENV
+          choco install python2 --params "/InstallDir:C:\"
+          python_dir=$(cat "C:\")
           python3_dir=$(cat "/proc/${{ matrix.cygreg }}/HKEY_LOCAL_MACHINE/SOFTWARE/Python/PythonCore/${PYTHON3_VER_DOT}${{ matrix.pyreg }}/InstallPath/@")
           echo "PYTHON_DIR=$python_dir" >> $GITHUB_ENV
```

Comment:
You can directly define an environment variable around line 373:
```yaml
      # Python 2
      PYTHON_VER: 27
      PYTHON_VER_DOT: '2.7'
      PYTHON_DIR: 'C:\Python27'
```

Then, you don't need to set the variable `python_dir` and you can remove the line `echo "PYTHON_DIR=$python_dir" >> $GITHUB_ENV`.

Or, the `choco install python2` command properly sets the registry, you should be able to leave the `python_dir=$(cat "/proc/ ... )` command as is.